core-bundle/src/InsertTag/ResolvedInsertTag.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\InsertTag;
+
+final class ResolvedInsertTag extends InsertTag
+{
+    /**
+     * @param list<InsertTagFlag> $flags
+     */
+    public function __construct(string $name, ResolvedParameters $parameters, array $flags)
+    {
+        parent::__construct($name, $parameters, $flags);
+    }
+
+    public function getParameters(): ResolvedParameters
+    {
+        if (!($parameters = parent::getParameters()) instanceof ResolvedParameters) {
+            throw new \TypeError(sprintf('%s(): Return value must be of type %s, got %s', __METHOD__, ResolvedParameters::class, $parameters::class));
+        }
+
+        return $parameters;
```

Comment:
<img width="778" alt="" src="https://github.com/contao/contao/assets/1192057/89ea82ba-4041-43cb-96c9-68264b2b25ca">

Reply:
Done in 8f68b497f21ca11e19afdac4d1154d51074c2016.